src/stk/molecular/molecules/building_block.py

```diff
@@ -644,15 +653,50 @@ def _normalize_placer_ids(self, placer_ids, functional_groups):
         """
 
         if placer_ids is not None:
-            return placer_ids
+            return frozenset(placer_ids)
 
         if functional_groups:
-            return tuple(flatten(
+            return frozenset(flatten(
                 functional_group.get_placer_ids()
                 for functional_group in functional_groups
             ))
 
-        return tuple(atom.get_id() for atom in self._atoms)
+        return frozenset(atom.get_id() for atom in self._atoms)
+
+    def _get_core_ids(
+        self,
+        functional_groups: Iterable[FunctionalGroup],
+    ) -> frozenset[int]:
+        """
+        Get the final *core* ids.
+
+        Parameters:
+
+            functional_groups:  The :class:`.FunctionalGroup` instances
+            of the building block.
+
+        Returns:
+
+            core_ids: Ids of atoms defining the core.
+
+        """
+
+        core_ids = []
```

Comment:
If you make this a set, rather than a list, your memory requirements while this function is running will be lower.